tests/native_api_tests/test_api_model.cpp

```diff
@@ -351,37 +375,183 @@ TEST_CASE("API Model") {
-            try {
-                model.calculate(options, batch_output_dataset, batch_update_dataset);
-                FAIL("Expected batch calculation error not thrown.");
-            } catch (PowerGridBatchError const& e) {
-                CHECK(e.error_code() == PGM_batch_error);
-                auto const& failed_scenarios = e.failed_scenarios();
-                CHECK(failed_scenarios.size() == 1);
-                CHECK(failed_scenarios[0].scenario == 1);
-                std::string const err_msg{failed_scenarios[0].error_message};
-                CHECK(err_msg.find("The id cannot be found:"s) != std::string::npos);
+            SUBCASE("Line bad line id") {
+                // wrong id
+                load_updates_id[1] = 999;
+                load_updates_buffer.set_value(PGM_def_update_sym_load_id, load_updates_id.data(), 1, -1);
+                // failed in batch 1
+                try {
+                    model.calculate(options, batch_output_dataset, batch_update_dataset);
+                    FAIL("Expected batch calculation error not thrown.");
+                } catch (PowerGridBatchError const& e) {
+                    CHECK(e.error_code() == PGM_batch_error);
+                    auto const& failed_scenarios = e.failed_scenarios();
+                    CHECK(failed_scenarios.size() == 1);
+                    CHECK(failed_scenarios[0].scenario == 1);
+                    std::string const err_msg{failed_scenarios[0].error_message};
+                    CHECK(err_msg.find("The id cannot be found:"s) != std::string::npos);
+                }
+                // valid results for batch 0
+                node_batch_output.get_value(PGM_def_sym_output_node_id, batch_node_result_id.data(), -1);
+                node_batch_output.get_value(PGM_def_sym_output_node_energized, batch_node_result_energized.data(), -1);
+                node_batch_output.get_value(PGM_def_sym_output_node_u, batch_node_result_u.data(), -1);
+                node_batch_output.get_value(PGM_def_sym_output_node_u_pu, batch_node_result_u_pu.data(), -1);
+                node_batch_output.get_value(PGM_def_sym_output_node_u_angle, batch_node_result_u_angle.data(), -1);
+                CHECK(batch_node_result_id[0] == 0);
+                CHECK(batch_node_result_energized[0] == 1);
+                CHECK(batch_node_result_u[0] == doctest::Approx(40.0));
+                CHECK(batch_node_result_u_pu[0] == doctest::Approx(0.4));
+                CHECK(batch_node_result_u_angle[0] == doctest::Approx(0.0));
+                CHECK(batch_node_result_id[1] == 4);
+                CHECK(batch_node_result_energized[1] == 0);
+                CHECK(batch_node_result_u[1] == doctest::Approx(0.0));
+                CHECK(batch_node_result_u_pu[1] == doctest::Approx(0.0));
+                CHECK(batch_node_result_u_angle[1] == doctest::Approx(0.0));
+            }
+        }
+    }
+
+    SUBCASE("Model update error optional id") {
```

Comment:
another potential source of error is if the size of a scenario for a specific component is not equal to the size of the input data of that component type. we should catch that as well

Reply:
Will be added via the row PR.